### DISCUSSION TITLE
Wallet connect not working on the Mobile Browsers

### DISCUSSION THREAD

**Discussion by sujal-empiric** [WalletConnect/walletconnect-monorepo#2953] [Developer Support]
The wallet connect module is not letting the user connect the wallet to a mobile browser 
When we select the trust wallet in the mobile it opens the Trust wallet but in the wallet, there is no pop-up for the request to connect. 
it's working fine for the browser but not on mobile 
I even tried the Metamask mobile but nothing happened.

**Comment by chadyj** [comment]
Hey we are happy to help, but first can you list what Trust and Metamask version you are using (make sure to upgrade to the latest), and what dapp you are trying to connect to.

**Comment by junaama** [comment]
I'm also experiencing the same thing with Metamask and Rainbow mobile wallets. I tested my dapp on multiple mobile browsers, Safari, Chrome, Brave and all have the same issue. It opens the mobile wallet app, but there's no connection request. On Metamask v7.2.0
@sujal-empiric Were you able to find a solution?

**Reply by dokocat** [reply]
the same thing has been happening for weeks, nobody wants to fix it I guess.

**Reply by glitch-txs** [reply]
hi, do you have a public repo where we can reproduce this issue?

Have you tested this on labs?
https://lab-walletconnect-modal.pages.dev/with-ethereum-provider

**Reply by dokocat** [reply]
Tried it. It can connect but if you try to sign a message it doesnt work. I'm on iOS using metamask. But I think its a metamask problem related to the chain ID.

**Comment by gintsmurans** [comment]
Hi!

Same thing here, tried labs example on iPad, metamask opens, but nothing happens. Same nothing is happening when trying Trust wallet. Tried in chrome and safari. Alternative link also does not work. Not sure if its metamask, wallet connect. Attached url it tried to open in firefox - if it helps anything. 

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/56369/f3cd9486-9f65-4261-bdf1-129d4313034c)

**Comment by mintbet7** [comment]
also had this happen when testing, I notice the first time the app launches, sometimes walletconnect loses track of the callback as the mobile app opens. this results in the wallet opening with no message, and the dapp hung waiting for a response. re-attempting the connection always works once the wallet app is open.

**Comment by VictorBlockchain** [comment]
is there a solution for this... been at this for hours... can't connect

**Comment by glitch-txs** [comment]
@VictorBlockchain please open an new discussion with more information. You can follow https://github.com/orgs/WalletConnect/discussions/3457
